### PR TITLE
Add integration tests using example files from apache/orc

### PR DESCRIPTION
**Pull request by progval**
Should I use `apache/orc` as a git submodule instead of copying data files here? I figured it's not worth the overhead of submodules, considering they only weigh 24MB.

Some tests are failing, I tried to annotate the reason when I understood. Some (commented with `// Why?`) look like actual bugs.

Resolves https://github.com/datafusion-contrib/datafusion-orc/issues/27

### CODE REVIEW

**Review by WenyXu — APPROVED** [review]
LGTM

File: tests/integration/main.rs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    if lines.len() < 1000 {
+        assert_eq!(lines, expected_lines);
+    } else {
+        // pretty_assertions consumes too much RAM and CPU on large diffs,
+        // and it's unreadable anyway
+        assert_eq!(lines[0..1000], expected_lines[0..1000]);
+        assert!(lines == expected_lines);
```
Comment (Jefffrey) [inline tests/integration/main.rs:68]: I wonder if we can parse the expected JSON into Arrow first with arrow_json then compare on RecordBatches

Assuming the schema inference works in our favour :thinking:

File: tests/integration/main.rs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+fn metaData() {
+    test_expected_file("TestOrcFile.metaData");
+}
+#[test]
+#[ignore] // Why?
+fn test1() {
+    test_expected_file("TestOrcFile.test1");
+}
+#[test]
+#[should_panic] // Incorrect timezone + representation differs
```
Comment (Jefffrey) [inline tests/integration/main.rs:93]: Thanks for leaving these annotations, gives us a goal to work towards

For all tests that fail we can just `#[ignore]` them and leave TODO comments to address these

**Review by Jefffrey — COMMENTED** [review]
Could you also add a README to the `tests/integration/data` directory just to indicate where the data came from? For easier viewing from the repo (instead of needing to track down the PR)

File: tests/integration/main.rs (hunk 1 above)
Comment (progval) [reply tests/integration/main.rs:68]: That would avoid the issue of different decimal representation.

However, it makes the tests a little unreliable, as they wouldn't detect data lost both by `arrow_json` and `datafusion_orc`. Probably not a big deal

Comment (Jefffrey) [reply tests/integration/main.rs:68]: That's a fair point. I guess we're a bit handicapped by the expected data being in JSON form which can make it harder for us to be rigorous. I'll raise an issue for exploring ways around this :+1:

**Comment by Jefffrey** [comment]
Thanks for this @progval :heart:

**Pull request merged**